### DISCUSSION TITLE
Could we render mixed code block, python and R for example, using the jupyter kernel?

### DISCUSSION THREAD

**Discussion by y9c** [quarto-dev/quarto-cli#2108] [Q&A]
Some ideas?
- knitr engine can render python code with the help from `reticulate` package. Can we convert R code in jupyter engine by `rpy2` package?
- Use the SOS engine? https://vatlab.github.io/sos-docs/
- specific multiple kernel for jupyter? `jupyter: ['python3', 'ir']` ?

**Comment by mcanouil** [comment]
Is it really a question, because you seem to have the answers. 🤔

**Reply by y9c** [reply]
Kind of... But none of these works for me. I don't know how to make quarto support multi kernels.

**Reply by mcanouil** [reply]
It's either Jupyter or knitr, you can't use both.
But knitr+reticulate should work as Jupyter+rpy2.
If that is not the case, please provide some reproducible example showing your use case and/or highlighting your issue.

**Reply by y9c** [reply]
Sorry for the confusion. I don't mean engines (Jupyter or knitr). I mean kernels in jupyter (python3 or irkernel)

**Reply by cderv** [reply]
For current state about several kernels, see existing discussions in 

* https://github.com/quarto-dev/quarto-cli/issues/135
* https://github.com/quarto-dev/quarto-cli/discussions/847#discussioncomment-2736658

Currently, it is one kernel only.

**Reply by y9c** [reply]
Thanks

**Comment by y9c** [comment]
![image](https://user-images.githubusercontent.com/5415510/186519549-3f348372-8a51-4ae1-bc15-5839305c44cd.png)

````md
---
title: "Quarto in jupyter"
format: html
jupyter: python3
---


```{python}
print("Hello python")
```

```{r}
print("Hello R")
```
````
